zebra-state/src/sled_state.rs

```diff
@@ -181,12 +184,21 @@ impl FinalizedState {
                 .coinbase_height()
                 .expect("valid blocks must have a height");
             self.commit_finalized(queued_block);
-            metrics::counter!("state.committed.block.count", 1);
-            metrics::gauge!("state.committed.block.height", height.0 as _);
+            metrics::counter!("state.finalized.committed.block.count", 1);
+            metrics::gauge!("state.finalized.committed.block.height", height.0 as _);
         }
 
+        if self.queued_by_prev_hash.is_empty() {
+            // use -1 as a sentinel value for "None", because 0 is a valid height
```

Comment:
Let's use an option rather than a sentinel value.

Reply:
@hdevalence I'm not sure if you've seen my original commit 1ddba8375a192d0efbd356946b5650799884e0c7, which uses `Option<Height>` for `self.max_queued_height`.

I'm happy to revert to that commit, if you and @yaahc agree that's the best way forward. Here's our previous discussion:
https://github.com/ZcashFoundation/zebra/pull/1262#discussion_r519002332

However using `-1` for the `metrics::gauge` is unavoidable, because the underlying `update_gauge` function only supports `i64`:
https://docs.rs/metrics/0.12.1/metrics/trait.Recorder.html#tymethod.update_gauge

(And ignoring the empty state is actually worse, because it incorrectly reports the previous maximum height.)